mediapipe/tasks/ios/vision/image_embedder/sources/MPPImageEmbedder.h

```diff
@@ -0,0 +1,264 @@
+ * `videoSettings` property.
+ *
+ * @param image A live stream image data of type `MPImage` on which embedding extraction is to be
+ * performed.
+ * @param timestampInMilliseconds The timestamp (in milliseconds) which indicates when the input
+ * image is sent to the image embedder. The input timestamps must be monotonically increasing.
+ *
+ * @return `true` if the image was sent to the task successfully, otherwise `false`.
+ */
+- (BOOL)embedAsyncImage:(MPPImage *)image
+    timestampInMilliseconds:(NSInteger)timestampInMilliseconds
+                      error:(NSError **)error
+    NS_SWIFT_NAME(embedAsync(image:timestampInMilliseconds:));
+
+/**
+ * Sends live stream image data of type `MPImage` to perform embedding extraction, cropped to the
+ * specified region of interest.. Rotation will be applied according to the `orientation` property
+ * of the provided `MPImage`. Only use this method when the `ImageEmbedder` is created with
+ * `.liveStream`.
+ *
+ * The object which needs to be continuously notified of the available results of image embedding
+ * extraction must confirm to `ImageEmbedderLiveStreamDelegate` protocol and implement the
+ * `imageEmbedder(_:didFinishEmbeddingWithResult:timestampInMilliseconds:error:)` delegate
+ * method.
+ *
+ * It's required to provide a timestamp (in milliseconds) to indicate when the input image is sent
+ * to the image embedder. The input timestamps must be monotonically increasing.
+ *
+ * This method supports embedding extraction on RGBA images. If your `MPImage` has a source type of
+ * `.pixelBuffer` or `.sampleBuffer`, the underlying pixel buffer must have one of the following
+ * pixel format types:
+ * 1. kCVPixelFormatType_32BGRA
+ * 2. kCVPixelFormatType_32RGBA
+ *
+ * If the input `MPImage` has a source type of `.image` ensure that the color space is RGB with an
+ * Alpha channel.
+ *
+ * If this method is used for embedding live camera frames using `AVFoundation`, ensure that you
+ * request `AVCaptureVideoDataOutput` to output frames in `kCMPixelFormat_32RGBA` using its
+ * `videoSettings` property.
+ *
+ * @param image A live stream image data of type `MPImage` on which embedding extraction is to be
+ * performed.
+ * @param timestampInMilliseconds The timestamp (in milliseconds) which indicates when the input
+ * image is sent to the image embedder. The input timestamps must be monotonically increasing.
+ * @param roi A `CGRect` specifying the region of interest within the given live stream image data
+ * of type `MPImage`, on which embedding extraction should be performed.
+ *
+ * @return `true` if the image was sent to the task successfully, otherwise `false`.
+ */
+- (BOOL)embedAsyncImage:(MPPImage *)image
+    timestampInMilliseconds:(NSInteger)timestampInMilliseconds
+           regionOfInterest:(CGRect)roi
+                      error:(NSError **)error
+    NS_SWIFT_NAME(embedAsync(image:timestampInMilliseconds:regionOfInterest:));
+
+- (instancetype)init NS_UNAVAILABLE;
```

Comment:
We should have a cosine similarity API as well: https://github.com/google/mediapipe/blob/master/mediapipe/tasks/python/vision/image_embedder.py#L339